tokenizer/src/lib.rs

```diff
@@ -201,13 +205,13 @@ impl<'a> TokenStream<'a> {
                         let (offset0, cp0) = chars.next().unwrap();
 
                         if cp0.is_id_start() {
-                            self.index += offset0 as u32 + 1;
+                            self.index = start + offset0 as u32;
 
                             for (offset, cp) in chars {
+                                self.index = start + offset as u32;
```

Comment:
This doesn't work when the last character of a source file belongs to an identifier. Since `offset` returns the offset where _the current character starts_, once we reach EOF `self.index` is off by one character. And in subsequent invocations of `next()`, `chars` in this loop is empty, which means `self.index` will be equal to `start`, causing an endless loop.